server.py

```diff
@@ -0,0 +1,61 @@
+import socket
+import sys
+
+ADDR = ('127.0.0.1', 8001) #port 0 may force os to find an open port
+
+def setup_server():
+    server_socket = socket.socket(
+    socket.AF_INET, socket.SOCK_STREAM, socket.IPPROTO_IP)
+    server_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+    server_socket.bind(ADDR)
+    server_socket.listen(1) #1 is backlog of connections it will hold onto
+    return server_socket
+
+
+def response_ok():
+    """ returns a response containg an inital response line, header, body"""
+    response_ok = """HTTP/1.1 200 OK\r\n\
+    DATE: Sun, 21 Jul 2001 23:32:15 GTM\r\n\
+    SERVER: Python/2.7.6\r\n\
+    \r\n\
+    """
+    return response_ok.encode('utf-8')
```

Comment:
check your assumptions here.  
1. You are using a multi-line string, that means that the newline characters your editor needs to make a new line of code are kept in the string.  So what sequence ends your lines in this response?
2. You have not imported `unicode_literals` from the `__future__` module, so your strings are bytestrings unless you explicitly mark them as unicode.  When you call encode on `response_ok`, what do you think happens?